build_tools/pkgci/external_test_suite/update_config_files.py

```diff
@@ -0,0 +1,135 @@
+#   3. Commit the modified config.json
+
+import argparse
+import json
+import logging
+import pyjson5
+
+logger = logging.getLogger(__name__)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Config file updater.")
+    parser.add_argument(
+        "--config-file",
+        default="",
+        required=True,
+        help="Path to a config JSON file (`pytest --config-files=config.json`) to update",
+    )
+    parser.add_argument(
+        "--log-file",
+        default="",
+        required=True,
+        help="Path to a log JSON file (`pytest --report-log=logs.json`) to read results from",
+    )
+    parser.add_argument(
+        "--log-level",
+        default="WARNING",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        type=str.upper,
+        help="Set the logging level",
+    )
+
+    args = parser.parse_args()
+    logging.basicConfig(level=args.log_level)
+    config_file = args.config_file
+
+    logger.info(f"Reading config from '{config_file}'")
+    with open(config_file, "r") as f:
+        config = pyjson5.load(f)
+
+        # Sanity check the config file structure before going any further.
+        def check_field(field_name):
+            if field_name not in config:
+                raise ValueError(
+                    f"config file '{config_file}' is missing a '{field_name}' field"
+                )
+
+        check_field("config_name")
+        check_field("iree_compile_flags")
+        check_field("iree_run_module_flags")
+
+    with open(args.log_file, "r") as f:
+        log_lines = f.readlines()
+
+    compile_failures = []
+    run_failures = []
+
+    for log_line in log_lines:
+        log_json = pyjson5.decode(log_line)
+
+        # Only look at test "call" events in the log.
+        if "when" not in log_json or log_json["when"] != "call":
+            continue
+
+        if log_json["outcome"] == "passed":
+            continue
+
+        # conftest.py adds extra info with `self.user_properties.append` for us
+        # to use. If that info is missing skip this log line.
+        if "user_properties" not in log_json or not log_json["user_properties"]:
+            logger.warning("Missing 'user_properties', ignoring log line")
+            continue
+        user_properties = log_json["user_properties"]
+
+        # TODO(scotttodd): handle multiple config files writing to one log file?
+
+        # Find the test directory path relative to the iree_tests root, since
+        # that is what our config.json uses to label tests for skipping or XFAIL.
+        relative_test_directory_name = ""
+        for user_property in user_properties:
+            if user_property[0] == "relative_test_directory_name":
+                relative_test_directory_name = user_property[1]
+        if not relative_test_directory_name:
+            logger.warning(
+                "Missing 'relative_test_directory_name' property, ignoring log line"
+            )
```

Comment:
https://github.com/iree-org/iree/actions/runs/9908826855/job/27375952971?pr=17885#step:10:22

```
WARNING:__main__:Missing 'relative_test_directory_name' property, ignoring log line
WARNING:__main__:Missing 'relative_test_directory_name' property, ignoring log line
...
  "expected_compile_failures": [],
  "expected_run_failures": []
```

This also needs the conftest.py changes from https://github.com/nod-ai/SHARK-TestSuite/pull/280/files#diff-cd2c3d64b04bf814e45e0b757ffa1c1625ed0b5371b22a4892131b76336a00c5

I'd either

A) update the test suite commit (hard because of commit sequencing)
B) keep using the old script but then manually add back `onnx/node/generated/` to the test cases
C) update the lists manually by looking at the logs

B should be easiest I think

Reply:
Ah I see, thanks